crates/release_plz_core/src/cargo.rs

```diff
@@ -150,6 +142,42 @@ async fn fetch_sparse_metadata(
     Ok(crate_data)
 }
 
+async fn request_for_sparse_metadata(
+    index: &SparseIndex,
+    crate_name: &str,
+    token: &Option<SecretString>,
+    // default is http2
```

Comment:
This parameter is mandatory so I don't think it has a "default" maybe we should move this comment down at line 154:

```
if force_http11 {
    // default is http2
    req = req.version(Version::HTTP_11);
}
```

wdyt?

Reply:
Also, maybe this could be an enum instead of a bool 👍

Reply:
I wanted to inform the caller of the method that HTTP 2.0 is enforced (by default), mainly because the doc of SparceIndex says HTTP 2 is recommended, but `make_cache_request` enforces http2 (that generates another error if http2 is not enabled on `client`).

I also thought about creating an enum (`HTTP_20`, `HTTP_11`) instead of the boolean. The goal was to document the parameter.

I'll do what you prefer.

Reply:
Instead of creating a new enum, I use http::Version.